rspace/src/main/scala/coop/rchain/rspace/history/package.scala

```diff
@@ -163,21 +200,37 @@ package object history {
             // we are in a situation where the new leaf shares some common prefix with the
             // existing leaf.
             case existingLeaf @ Leaf(ek, _) if key != ek =>
```

Comment:
Also somewhat fishy, this code path is also unexercised after these changes.

See the [current report](https://codecov.io/gh/rchain/rchain/pull/993/src/rspace/src/main/scala/coop/rchain/rspace/history/package.scala).

Reply:
I was so fond of this piece of code :sob:
The path is no longer taken because of changes in getParents. 
The trie will always contain Skip nodes or Node nodes before it runs into a Leaf and it will be able to stop the traversal there.
(It used to be the case that Leaf nodes were the end of the path. Now Skip nodes have taken this role over.)